crates/control/src/filtered_game_controller_state_timer.rs

```diff
@@ -0,0 +1,159 @@
+use std::time::SystemTime;
+
+use color_eyre::Result;
+use context_attribute::context;
+use framework::MainOutput;
+use serde::{Deserialize, Serialize};
+use spl_network_messages::Penalty;
+use types::{
+    cycle_time::CycleTime, filtered_game_controller_state::FilteredGameControllerState,
+    last_filtered_game_controller_state_change::LastFilteredGameControllerStateChanges,
+    players::Players,
+};
+
+#[derive(Deserialize, Serialize)]
+pub struct FilteredGameControllerStateTimer {
+    last_filtered_game_controller_state: FilteredGameControllerState,
+    filtered_game_controller_state_changes: LastFilteredGameControllerStateChanges,
+}
+
+#[context]
+pub struct CreationContext {}
+
+#[context]
+pub struct CycleContext {
+    cycle_time: Input<CycleTime, "cycle_time">,
+    filtered_game_controller_state:
+        RequiredInput<Option<FilteredGameControllerState>, "filtered_game_controller_state?">,
+}
+
+#[context]
+#[derive(Default)]
+pub struct MainOutputs {
+    pub last_filterd_game_controller_state_changes:
+        MainOutput<Option<LastFilteredGameControllerStateChanges>>,
```

Comment:
remove newlines

Reply:
I think I'm done, but I'm not sure which newline you mean

Reply:
oh sorry, github  view trolled me :D